site/docs/types/type_classes.md

```diff
@@ -44,20 +44,50 @@ Compound type classes are type classes that need to be configured by means of a
 
 ## User-Defined Types
 
-User-defined type classes can be created using a combination of pre-defined types. User-defined types are defined as part of [simple extensions](../extensions/index.md#simple-extensions). An extension can declare an arbitrary number of user defined extension types. Once a type has been declared, it can be used in function declarations.
+User-defined type classes are defined as part of [simple extensions](../extensions/index.md#simple-extensions). An extension can declare an arbitrary number of user-defined extension types. Once a type has been declared, it can be used in function declarations.
 
-A YAML example of an extension type is below:
+For example, the following declares a type named `point` (namespaced to the associated YAML file) and two scalar functions that operate on it.
 
-```yaml
-name: point
-structure:
-  longitude: i32
-  latitude: i32
 ```
+---
+types:
```

Comment:
Could this also be part of a yaml code block like the previous example was?

Reply:
Done, I think?